### DISCUSSION TITLE
Boundary wall conditions problem

### DISCUSSION THREAD

**Discussion by appheadspace** [firemodels/fds#13663] [Q&A] [answered]
I am solving a natural convection problem in closed cubic cavity in FDS. I am giving hot and cold wall temps as 50C and 10C. But it is not showing the temps at t=0 and also they are not reaching that temp even after 2hrs simulation. Pls help!!!

**My input file:**

&HEAD CHID='chid' TITLE='simulation' /

&MESH IJK=20,20,20, XB=0,0.75,-0.375,0.375,0,0.75 /

&TIME T_END=7200/

&DUMP DT_DEVC=60,  DT_RESTART = 60/

&MISC STRATIFICATION=F, SIMULATION_MODE='LES'/

&RADI RADIATION=.TRUE./

&SPEC ID='LJ AIR', SPECIFIC_HEAT=1.007, CONDUCTIVITY=0.02588, VISCOSITY=1.872E-5, BACKGROUND=.TRUE./

&VENT PBX = 0, 		SURF_ID='hot_wall'  / left face
&VENT PBX = 0.75, 	SURF_ID='cold_wall' / right face
&VENT PBY = -0.375, SURF_ID='insulated' / back face
&VENT PBY = 0.375, 	SURF_ID='insulated' / front face
&VENT PBZ=0, 		SURF_ID='insulated' / bottom face
&VENT PBZ=0.75, 	SURF_ID='insulated' / top face

&SURF ID = 'hot_wall', 		COLOR = 'RED',  	TMP_FRONT = 50, 	TAU_T=0., EMISSIVITY=0.9 /
&SURF ID = 'cold_wall', 	COLOR = 'BLUE',  	TMP_FRONT = 10, 	TAU_T=0., EMISSIVITY=0.9 /
&SURF ID = 'insulated', 	COLOR = 'GRAY', 	HEAT_TRANSFER_COEFFICIENT=0./

&INIT XB=0,0.75,-0.375,0.375,0,0.75, 	TEMPERATURE=30./

&DEVC XYZ=0,0,0.375, 			IOR=1, 	QUANTITY='TEMPERATURE', ID='HotWall_Temp'/
&DEVC XYZ=0.75,0,0.375, 		IOR=-1, QUANTITY='TEMPERATURE', ID='ColdWall_Temp'/
&DEVC XYZ=0.375,-0.375,0.375, 	IOR=2, 	QUANTITY='TEMPERATURE', ID='BackWall_Temp'/
&DEVC XYZ=0.375,0.375,0.375, 	IOR=-2, QUANTITY='TEMPERATURE', ID='FrontWall_Temp'/
&DEVC XYZ=0.375,0,0, 			IOR=3, 	QUANTITY='TEMPERATURE', ID='BottomWall_Temp'/
&DEVC XYZ=0.375,0,0.75, 		IOR=-3, QUANTITY='TEMPERATURE', ID='TopWall_Temp'/

&DEVC XYZ=0,0,0.375, 			IOR = 1,  	QUANTITY='CONVECTIVE HEAT FLUX', ID='HotWall_ConvectiveHeatFlux'/
&DEVC XYZ=0.75,0,0.375, 		IOR = -1, 	QUANTITY='CONVECTIVE HEAT FLUX', ID='ColdWall_ConvectiveHeatFlux'/
&DEVC XYZ=0.375,-0.375,0.375, 	IOR=2, 		QUANTITY='CONVECTIVE HEAT FLUX', ID='BackWall_ConvectiveHeatFlux'/
&DEVC XYZ=0.375,0.375,0.375, 	IOR=-2, 	QUANTITY='CONVECTIVE HEAT FLUX', ID='FrontWall_ConvectiveHeatFlux'/
&DEVC XYZ=0.375,0,0, 			IOR=3, 		QUANTITY='CONVECTIVE HEAT FLUX', ID='BottomWall_ConvectiveHeatFlux'/
&DEVC XYZ=0.375,0,0.75, 		IOR=-3, 	QUANTITY='CONVECTIVE HEAT FLUX', ID='TopWall_ConvectiveHeatFlux'/

&DEVC XYZ=0,0,0.375, 			IOR = 1,  	QUANTITY='HEAT TRANSFER COEFFICIENT', ID='HotWall_HTC'/
&DEVC XYZ=0.75,0,0.375, 		IOR = -1, 	QUANTITY='HEAT TRANSFER COEFFICIENT', ID='ColdWall_HTC'/
&DEVC XYZ=0.375,-0.375,0.375, 	IOR=2, 		QUANTITY='HEAT TRANSFER COEFFICIENT', ID='BackWall_HTC'/
&DEVC XYZ=0.375,0.375,0.375, 	IOR=-2, 	QUANTITY='HEAT TRANSFER COEFFICIENT', ID='FrontWall_HTC'/
&DEVC XYZ=0.375,0,0, 			IOR=3, 		QUANTITY='HEAT TRANSFER COEFFICIENT', ID='BottomWall_HTC'/
&DEVC XYZ=0.375,0,0.75, 		IOR=-3, 	QUANTITY='HEAT TRANSFER COEFFICIENT', ID='TopWall_HTC'/

&DEVC XYZ=0.375,0,0.375,  QUANTITY='TEMPERATURE', 			ID='Core_MidPoint_Temp' /
&DEVC XYZ=0.375,0,0.375,  QUANTITY='BACKGROUND PRESSURE',   ID='Core_MidPoint_Pressure' /
&DEVC XYZ=0.375,0,0.375,  QUANTITY='DENSITY', 				ID='Core_MidPoint_Density' /

&TAIL /

**Devc file:**
[devc.csv](https://github.com/user-attachments/files/17586241/devc.csv)

**Comment by Er9y714** [comment]
Can you explain the issue in more detail? What values do you expect at which devices which they do not reach?
You can also add slices with DT_SLCF=1 to see what is happening.

**Reply by appheadspace** [reply]
<img width="535" alt="image" src="https://github.com/user-attachments/assets/4f6b3b12-718d-49da-b4b2-c338c1c7e104">

Thank you for ur reply! The initial temperature I gave for hot and cold wall is 50 and 10C. But at t=0, it is 30C (temp given in INIT). It should be 50 and 10 instead of 30.

**Comment by drjfloyd** [comment]
TEMPERATURE is a gas phase quantity.  It is not going to give you the temperature of the solid surface. See the table of solid phase output quantities in the User Guide.

You don't need to define the specific heat, conductivity, etc. on SPEC. LJ AIR is a predefined species and already has those defined.

**Reply by appheadspace** [reply]
Thank you for ur reply! Even in that case(as u said), the temp of inside air must change according to wall boundary conditions. But the temperature inside cavity is around 30C only.

**Reply by appheadspace** [reply]
![download](https://github.com/user-attachments/assets/d99f3ba7-7e1b-4f79-af63-84ecb8c9e6a5)

**Reply by Er9y714** [reply]
You can add a slice of TEMPERATURE to monitor the temperature development. Since it is a natural convection, it would take time to heat the gas. 
Also try setting DT_DEVC lower (e.g. 0.1 s) to monitor the temperatures near T=0.

**Reply by appheadspace** [reply]
Can u pls kindly tell me what I would know if DT_DEVC <0.1 sec? bcoz I need the gas to be around 50C near hot wall and 10C near cold wall and it is not reaching even after 60 sec or 2 hrs.. I am new to FDS. Please explain.

**Reply by Er9y714** [reply, marked as answer]
DT_DEVC is the duration of time the output is averaged. To accurately follow the output in shorter time intervals, DT_DEVC should be small (in your model it is 60 seconds). 
 
 
try adding this device line to monitor wall temperature:
&DEVC ID='Wall temp Hot wall', QUANTITY='WALL TEMPERATURE', XYZ=0.0,0.0,0.0, IOR=1/
and replace DUMP line with this:
&DUMP DT_SLCF=1 DT_DEVC=0.1/


If you add this line, you can monitor the temperature profile in time:
&SLCF QUANTITY='TEMPERATURE', ID='Slice', PBY=0.01875/

**Reply by drjfloyd** [reply]
The hot wall side is higher. The cold wall side is lower. The average temperature inside your domain should be near 30 C.  At the mid height in your domain you should expect that on the hot wall side the bulk fluid should be heated halfway from the coldest temperature to its hotest, and on the cold wall side the fluid should be cooled halfway from its hottest temperature to its lowest. Put in a slice file of temperature at Y=0 and look at the results in Smokeview.

**Reply by appheadspace** [reply]
@Er9y714 , I have tried the changes you have suggested. "WALL TEMPERATURE" instead of "TEMPERATURE" worked!! it is now showing the hot and cold wall temps as 50 and 10C. But temp PBX = 0.01875 and PBY = 0.01875 is showing near 30 only (at mid height z = 0.375m). Kindly tell me what to do to get near hot wall temp ~50C (and near cold wall temp ~10C).

<img width="386" alt="image" src="https://github.com/user-attachments/assets/720ffcfb-b3f0-40a8-96ac-1ee406b2f682">

**Reply by appheadspace** [reply]
@drjfloyd , Thanks for your reply! PFB my smv snapshot. I need to plot graphs from the results but temperatures inside the cavity are near 50C only. 

<img width="449" alt="image" src="https://github.com/user-attachments/assets/90514f3e-f099-49f2-82f6-abf83400ccea">

**Reply by Er9y714** [reply]
> @Er9y714 , I have tried the changes you have suggested. "WALL TEMPERATURE" instead of "TEMPERATURE" worked!! it is now showing the hot and cold wall temps as 50 and 10C. But temp PBX = 0.01875 and PBY = 0.01875 is showing near 30 only (at mid height z = 0.375m). Kindly tell me what to do to get near hot wall temp ~50C (and near cold wall temp ~10C).
> 
If you refine the mesh, you might observe higher temperature near the hot wall. Note that the gas temperature you measure with devices gives the temperature of the cell within. So if you set the mesh to 2 mm, place the device in the center of the nearest cell to the wall.

**Reply by drjfloyd** [reply]
Your grid cells are almost 4 cm. You are not close to resolving the boundary layer layer where near 50 C temperatures would exist for this geometry. You are going to need an order of magnitude more resolution to resolve the boundary layer.

**Reply by appheadspace** [reply]
@drjfloyd @Er9y714 . If I'm trying to create finer mesh near the wall, the simulation is taking very long time. Can u pls suggest how to do 2D simulation for this problem? I've seen in Matteo Pachera paper that he did 2D simulation with 20x20 mesh (400 cells) in FDS for the same problem. Kindly guide me towards that.

**Reply by Er9y714** [reply]
Have you checked the FDS User guide about how to set up 2D domain?

**Reply by appheadspace** [reply]
Yes. I wanted to start with that approach only. But haven't found anything reg that. So proceeded with 3D🥲

**Reply by Er9y714** [reply]
6.3.2 Two-Dimensional and Axially-Symmetric Calculations
